map_model/src/objects/lane.rs

```diff
@@ -287,40 +289,31 @@ impl Lane {
         }
         // TODO More warnings if this fails
         let part = parts[lanes.iter().position(|l| *l == self.id)?];
+
         // TODO Probably the target lane should get marked as LaneType::Bus
-        if part == "no" || part == "none" || part == "yes" || part == "psv" || part == "bus" {
+        if part == "yes" || part == "psv" || part == "bus" {
             return None;
         }
-        // Empty means no restrictions
-        if part.is_empty() {
-            return None;
+
+        // These both mean that physically, there's no marking saying what turn is valid. In
+        // practice, this seems to imply straight is always fine, and right/left are fine unless
+        // covered by an explicit turn lane.
+        if part == "" || part == "none" {
+            let all_explicit_types: BTreeSet<TurnType> = parts
+                .iter()
+                .flat_map(|part| part.split(';').flat_map(parse_turn_type_from_osm))
+                .collect();
+            let mut implied = BTreeSet::new();
+            implied.insert(TurnType::Straight);
+            for tt in vec![TurnType::Left, TurnType::Right] {
+                if !all_explicit_types.contains(&tt) {
+                    implied.insert(tt);
+                }
+            }
+            return Some(implied);
         }
-        Some(
-            part.split(';')
-                .flat_map(|s| match s {
-                    "left" | "left\\left" => vec![TurnType::Left],
```

Comment:
The astute reviewer will notice that I got rid of this `left\\left` thing -- I don't remember where I encountered this, and if I see it again, I'll just fix in OSM.

And also I'm mapping `reverse` to `TurnType::UTurn` now that we have it, but I haven't hunted around to find examples of this.